### PR TITLE
Add io.github.trytonvanmeer.DungeonJournal

### PR DESCRIPTION
**Pull request by tryton-vanmeer**

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build io.github.trytonvanmeer.DungeonJournal

File: io.github.trytonvanmeer.DungeonJournal.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+    ],
+    "build-options": {
```
Comment (bilelmoussaoui) [inline io.github.trytonvanmeer.DungeonJournal.json]: Those build options are now default in fdo 18.08. Not needed.

File: io.github.trytonvanmeer.DungeonJournal.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                {
+                    "type" : "git",
+                    "url" : "https://source.puri.sm/Librem5/libhandy",
+                    "branch" : "v0.0.10"
+                }
+            ]
+        },
+        {
+            "name": "dungeonjournal",
+            "builddir": true,
```
Comment (bilelmoussaoui) [inline io.github.trytonvanmeer.DungeonJournal.json]:
```suggestion
```
it's set by default

File: io.github.trytonvanmeer.DungeonJournal.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            "buildsystem": "meson",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/tryton-vanmeer/DungeonJournal"
```
Comment (bilelmoussaoui) [inline io.github.trytonvanmeer.DungeonJournal.json]: We need a commit or tarball here. Flathub builds must be reproducible

**Comment by bilelmoussaoui** [comment]
The icon is either not installed or installed with an incorrect name
```
FB: Running: flatpak build --die-with-parent --nofilesystem=host /srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/rofiles/rofiles-zFOkMi appstream-compose --prefix=/app --origin=flatpak --basename=io.github.trytonvanmeer.DungeonJournal io.github.trytonvanmeer.DungeonJournal
Processing application io.github.trytonvanmeer.DungeonJournal
Error loading desktop file: Failed to find icon: Failed to find icon com.github.trytonvanmeer.DungeonJournal
Error: ERROR: appstream-compose failed: Child process exited with code 1
 ```

**Comment by tryton-vanmeer** [comment]
Thanks, I'll make the appropriate changes. I caught the mistake I made with the wrong id in the creating the about dialog but must have missed it in the desktop file.

**Comment by nedrichards** [comment]
@tryton-vanmeer you can use this command as well to generate a fresh test build if you do an update btw.

bot, build io.github.trytonvanmeer.DungeonJournal

**Comment by tryton-vanmeer** [comment]
bot, build io.github.trytonvanmeer.DungeonJournal

**Comment by tryton-vanmeer** [comment]
bot, build io.github.trytonvanmeer.DungeonJournal

File: io.github.trytonvanmeer.DungeonJournal.json [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                "-Dintrospection=enabled",
+                "-Dexamples=false",
+                "-Dtests=false"
+
+            ],
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://source.puri.sm/Librem5/libhandy",
+                    "branch" : "v0.0.10"
```
Comment (bilelmoussaoui) [inline io.github.trytonvanmeer.DungeonJournal.json]: Please add a commit too. Flathub is about reproducible builds.

**Comment by bilelmoussaoui** [comment]
bot, build io.github.trytonvanmeer.DungeonJournal

**Comment by tryton-vanmeer** [comment]
bot, build io.github.trytonvanmeer.DungeonJournal

File: io.github.trytonvanmeer.DungeonJournal.json [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                "-Dintrospection=enabled",
+                "-Dexamples=false",
+                "-Dtests=false"
+
+            ],
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://source.puri.sm/Librem5/libhandy",
+                    "branch" : "2d777677352d037b6f5cc24d9c1c8d9a74ac0ded"
```
Comment (bilelmoussaoui) [inline io.github.trytonvanmeer.DungeonJournal.json]:
```suggestion
                    "commit" : "2d777677352d037b6f5cc24d9c1c8d9a74ac0ded"
```

File: io.github.trytonvanmeer.DungeonJournal.json [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            ]
+        },
+        {
+            "name": "dungeonjournal",
+            "buildsystem": "meson",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/tryton-vanmeer/DungeonJournal",
+                    "branch": "3e85aed5b613e7641f80584d9d556dbc931d352a"
```
Comment (bilelmoussaoui) [inline io.github.trytonvanmeer.DungeonJournal.json]:
```suggestion
                    "commit": "3e85aed5b613e7641f80584d9d556dbc931d352a"
```

**Comment by tryton-vanmeer** [comment]
Sorry about that. I was referencing another manifest and it used `branch` when specifying the commit. Should have figured to use `commit`.

**Comment by tryton-vanmeer** [comment]
bot, build io.github.trytonvanmeer.DungeonJournal

**Comment by bilelmoussaoui** [comment]
I would disable arm builds using flathub.json & investigate that later.

**Comment by tryton-vanmeer** [comment]
bot, build io.github.trytonvanmeer.DungeonJournal

**Comment by bilelmoussaoui** [comment]
It works fine other than this 
```
(dungeonjournal:2): GLib-CRITICAL **: 23:32:43.878: g_variant_new_string: assertion 'string != NULL' failed
```
Might worth looking at the cause of that.

**Comment by bilelmoussaoui** [comment]
Repository has been created: https://github.com/flathub/io.github.trytonvanmeer.DungeonJournal


Thanks!

**Pull request closed without merging**